tests/integration/pilot/ingress_test.go

```diff
@@ -538,7 +538,7 @@ spec:
 						Headers: map[string][]string{
 							"Host": {"server"},
 						},
-						Validator: echo.ExpectCode("404"),
+						Validator: echo.ExpectError(),
```

Comment:
seems like we may want error or 404? 

I think this only works under the assumption there are zero other Gateways in the entire cluster. Not sure how safe that assumption is...

Reply:
Makes sense, added `echo.Or` and used here.